Vector.hpp

```diff
@@ -0,0 +1,253 @@
+     */
+    void insert(size_t index, const T& value)
+    {
+        if (index > elements_Size)
+        {
+            throw std::out_of_range("Error: Index out of range.");
+        }
+
+        if (elements_Size >= elements_Capacity)
+        {
+            inc_size();
+        }
+
+        for (size_t i = elements_Size; i > index; --i)
+        {
+            m_Data[i] = std::move(m_Data[i - 1]);
+        }
+
+        m_Data[index] = value;
+        ++elements_Size;
+    }
+
+    /**
+     * @brief Reverse the order of elements in the vector.
+     */
+    void reverse() {
+        T temp;
+        for (size_t i = 0; i < elements_Size / 2; ++i) {
+            temp = m_Data[i];
+            m_Data[i] = m_Data[elements_Size - i - 1];
+            m_Data[elements_Size - i - 1] = temp;
+        }
+    }
+
+    /**
+     * @brief Add an element to the end of the vector by constructing it in-place.
+     *
+     * @param Args The types of arguments to forward to the constructor of T.
+     * @param args The arguments to forward to the constructor of T.
+     */
+    template<typename... Args>
+    void emplace_back(Args&&... args)
+    {
+        if (elements_Size >= elements_Capacity)
+        {
+            inc_size();
+        }
+        new(&m_Data[elements_Size++]) T(std::forward<Args>(args)...);
+    }
+
+    /**
+     * @brief Access an element at a specified position.
+     *
+     * @param index The position of the element.
+     * @return The element at the specified position.
+     * @throw std::out_of_range if the index is out of range.
+     */
+    const T& operator[](size_t index) const
+    {
+        if (index >= elements_Size)
+        {
+            throw std::out_of_range("Error: Index out of range.");
+        }
+        return m_Data[index];
+    }
+
+    /**
+     * @brief Access an element at a specified position with bounds checking.
+     *
+     * @param index The position of the element.
+     * @return The element at the specified position.
+     * @throw std::out_of_range if the index is out of range.
+     */
+    const T& at(size_t index) const
+    {
+        if (index >= elements_Size)
+        {
+            throw std::out_of_range("Error: Index out of range.");
+        }
+        return m_Data[index];
+    }
+
+    /**
+     * @brief Compare two vectors for equality.
+     *
+     * @param other The vector to compare with.
+     * @return true if the vectors are equal, false otherwise.
+     */
+    bool operator==(const Vector& other) const
+    {
+        if (elements_Size != other.elements_Size)
+        {
+            return false;
+        }
+        for (size_t i = 0; i < elements_Size; ++i)
+        {
+            if (m_Data[i] != other.m_Data[i])
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+private:
+    T* m_Data = nullptr;              ///< Pointer to the dynamically allocated array of elements.
```

Comment:
member variables should be initialized by the constructor and not on their own.
also convention states that they should start with a `_` so you'll have `_data`

Reply:
also member variables should be `pascalCase` and not `snake_case`